step-capstone/src/components/Sidebars/HandleClickedTimePoint.js

```diff
@@ -0,0 +1,176 @@
+export const getEmptySlots = (startOfDisplayDate, endOfDisplayDate, displayItems) => {
+  let emptySlots = [];
+
+  if (displayItems.length == 0) {
+    emptySlots.push({
+      startDate: startOfDisplayDate,
+      endDate: endOfDisplayDate,
+    });
+    return emptySlots;
+  } else {
+    if (displayItems[0].startDate > startOfDisplayDate) {
+      emptySlots.push({
+        startDate: startOfDisplayDate,
+        endDate: displayItems[0].startDate,
+      });
+    }
+  }
+
+  for (var i = 1; i <= displayItems.length - 1; i++) {
+    let prev = displayItems[i - 1];
+    let cur = displayItems[i];
+
+    emptySlots.push({ startDate: prev.endDate, endDate: cur.startDate });
+  }
+
+  if (displayItems[displayItems.length - 1].endDate < endOfDisplayDate) {
+    emptySlots.push({
+      startDate: displayItems[displayItems.length - 1].endDate,
+      endDate: endOfDisplayDate,
+    });
+  }
+
+  return emptySlots;
+};
+
+const contains = (interval, timePoint) => {
+  // return whether timeRange from startDate to endDate contains timePoint
```

Comment:
Nit: move this comment to be above the function for consistency within the file.